homeassistant/components/nws/weather.py

```diff
@@ -243,6 +244,7 @@ def forecast(self) -> list[Forecast] | None:
                 ATTR_FORECAST_DETAILED_DESCRIPTION: forecast_entry.get(
                     "detailedForecast"
                 ),
+                ATTR_FORECAST_SUMMARY: forecast_entry.get("shortForecast"),
```

Comment:
Let's make sure we exclude it from the recorder database.

Reply:
Are you referring to `entity_registry_enabled_default`?  If so this is set to only have one such entity by default.  Otherwise, I'm not sure how to exclude individual forecast data from the recorder, so if you could provide a pointer, it would be helpful.

Reply:
No I'm referencing implementing the recorder platform and exclude this attribute from being recoreded.

Reply:
I don't see any developer documentation and there aren't many integrations that implement this for guidance.  When I check the weather integration, it is already including the `forecast` key in the `exclude_attributes`.  Does this already exclude this nested key?

https://github.com/home-assistant/core/blob/7f616b0d44e18416b0802cb30322eead52d7840c/homeassistant/components/weather/recorder.py#L9-L12

Reply:
Oh now I see, that I missed in my original looks this PR. This is about the forecast attribute. Ignore my recorder comment.

The forecast attribute is a strict set, you can't / may not add new properties to it; this is the list defined:
https://github.com/home-assistant/core/blob/7f616b0d44e18416b0802cb30322eead52d7840c/homeassistant/components/weather/__init__.py#L140-L167

Reply:
We are not adding new properties to the forecast attribute.  The NWS component brings forward a weather entity called daynight ( weather.XXXX_daynight where XXXX is the NWS station ID ).  This PR add an attribute to that entity for a short concise forecast text string.  The underlying pyNWS library presents it, we are just exposing it as an additional attribute in the _daynight entity.

Here's a screen shot:
![image](https://github.com/home-assistant/core/assets/3859141/e62f36fb-4988-403c-b71d-f887a46c0a14)

Reply:
The forecast attribute of weather is not supposed to have an `short_forecast` attribute. It should also not have a `detailed_description` attribute in the forecast either (which should be removed).

See my previous response on the properties that are allowed to be in the forecast dictionary items in the list.

Reply:
Unresolved this review comment.